app/models/report.rb

```diff
@@ -14,4 +21,16 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def create_mentions
+    mentioned_report_ids = content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.uniq
+    mentioned_report_ids.map(&:to_i).each do |id|
+      active_mentions.create!(mentioned_report_id: id, mentioning_report_id: self.id)
+    end
+  end
+
+  def update_mentions
+    self.active_mentions.find_each(&:destroy!)
```

Comment:
```suggestion
    self.active_mentions. destroy_all
```

の方がシンプルかなーと思います。

細かいことをいえば destroy_all だと削除に失敗しても気づけない可能性がある、という違いはありますが、アプリケーションの要件上、この削除に失敗するケースはほぼないと思うので、 destroy_all で大丈夫かなーと思います。

Reply:
ありがとうございます！
要件によっては`destroy_all`がベターな場合もあるんですね。
ここに限らず、「要件に照らして不要な機能がないか？」「不要な機能を削げばより読みやすくならないか？」を考えるようにしたいと思います。